File: TesteAndroid/app/src/main/java/com/accenture/android/app/testeandroid/presentation/login/LoginContract.java
package com.accenture.android.app.testeandroid.presentation.login;

import com.accenture.android.app.testeandroid.presentation.BasePresenter;
import com.accenture.android.app.testeandroid.presentation.BaseView;

/**
 * Created by Denis Magno on 09/07/2020.
 * denis_magno16@hotmail.com
 */
interface LoginContract {
    interface Presenter extends BasePresenter {
        void efetuarLogin(String user, String password);
    }

    interface View extends BaseView<Presenter> {
        void setLoading();

        void unsetLoading();

        void setContent();

        void unsetContent();

        void setFeedback(String message);

        void setError(String message);

        void unsetError();

        void setLoginRecente(String name);

        void navigateToMainActivity();
    }
}

File: App/presentation/src/main/java/com/santander/app/feature/login/LoginContract.kt
package com.santander.app.feature.login

import com.santander.app.core.ui.base.BasePresenter
import com.santander.app.core.ui.base.BaseView

object LoginContract {

    interface View: BaseView {
        fun displayUser(user: String)
        fun onAuthenticationSuccess()
    }

    interface Presenter: BasePresenter<View> {
        fun doLogin(user: String, password: String)
    }
}